store/postgres/src/store.rs

```diff
@@ -209,59 +328,74 @@ impl Store {
         Ok(())
     }
 
-    /// Handles entity changes emitted by Postgres.
-    fn handle_entity_changes(
-        &self,
-        entity_changes: Box<Stream<Item = EntityChange, Error = ()> + Send>,
-    ) {
+    /// Handles change events emitted by Postgres.
+    fn handle_store_events(&self, store_events: Box<Stream<Item = StoreEvent, Error = ()> + Send>) {
         let logger = self.logger.clone();
         let subscriptions = self.subscriptions.clone();
+        let dirty_blocks = self.dirty_blocks.clone();
 
-        tokio::spawn(entity_changes.for_each(move |change| {
-            trace!(logger, "Received entity change event";
+        tokio::spawn(store_events.for_each(move |event| {
+            match &event {
+                StoreEvent::Change(change) => {
+                    trace!(logger, "Received entity change event";
                            "subgraph_id" => change.subgraph_id.to_string(),
                            "entity_type" => &change.entity_type,
                            "entity_id" => &change.entity_id);
+                }
+                StoreEvent::Tick(tick) => {
+                    trace!(logger, "Received block tick"; "block" => tick.block.to_string());
+                }
+            }
+
+            // Record which BlockTick subscriptions have had 'interesting'
+            // changes and are therefore dirty
+            for (id, subscription) in subscriptions.read().unwrap().iter() {
+                if subscription.is_dirty(&event) {
+                    dirty_blocks.insert(id.clone());
+                }
+            }
 
             // Obtain IDs and senders of subscriptions matching the entity change
             let matches = subscriptions
                 .read()
                 .unwrap()
                 .iter()
-                .filter(|(_, subscription)| {
-                    subscription
-                        .entities
-                        .contains(&(change.subgraph_id.clone(), change.entity_type.clone()))
-                })
-                .map(|(id, subscription)| (id.clone(), subscription.sender.clone()))
+                .filter(|(id, subscription)| subscription.matches(&event, id, &dirty_blocks))
```

Comment:
A comment here would be nice too: What does it mean if a subscription matches here?